crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -0,0 +1,1089 @@
+                {
+                    return Ordering::Greater
+                }
+                (Some(a_char), Some(b_char))
+                    if !a_char.is_alphanumeric() && b_char.is_alphanumeric() =>
+                {
+                    return Ordering::Less
+                }
+                (Some(a_char), Some(b_char)) if a_char == '[' && b_char.is_alphanumeric() => {
+                    return Ordering::Greater
+                }
+                (Some(a_char), Some(b_char)) if a_char.is_alphanumeric() && b_char == '[' => {
+                    return Ordering::Less
+                }
+                (Some(a_char), Some(b_char)) => return a_char.cmp(&b_char),
+                (None, None) => return Ordering::Equal,
+                (Some(_), None) => return Ordering::Greater,
+                (None, Some(_)) => return Ordering::Less,
+            }
+        }
+    });
+}
+
+fn take_numeric(iter: &mut Chars, first: char) -> u32 {
+    let mut sum = first.to_digit(10).unwrap();
+    for c in iter.by_ref() {
+        if let Some(digit) = c.to_digit(10) {
+            sum = sum * 10 + digit;
+        } else {
+            break;
+        }
+    }
+    sum
+}
+
+fn extract_text_between_spans(source_text: &str, current_span: Span, next_span: Span) -> &str {
+    let cur_span_end = current_span.end as usize;
+    let next_span_start = next_span.start as usize;
+    &source_text[cur_span_end..next_span_start]
+}
+
+fn extract_property_key(prop_text: &str) -> &str {
+    let trimmed = prop_text.trim();
+    let before_colon = trimmed.split(':').next().unwrap_or(trimmed);
+    let without_quotes = before_colon.split('"').next().unwrap_or(before_colon);
+    without_quotes.split('(').next().unwrap_or(before_colon)
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("var obj = {'':1, [``]:2}", Some(serde_json::json!([]))), // { "ecmaVersion": 6 },
+        ("var obj = {[``]:1, '':2}", Some(serde_json::json!([]))), // { "ecmaVersion": 6 },
+        ("var obj = {'':1, a:2}", Some(serde_json::json!([]))),
+        ("var obj = {[``]:1, a:2}", Some(serde_json::json!([]))), // { "ecmaVersion": 6 },
+        ("var obj = {_:2, a:1, b:3} // default", Some(serde_json::json!([]))),
+        ("var obj = {a:1, b:3, c:2}", Some(serde_json::json!([]))),
+        ("var obj = {a:2, b:3, b_:1}", Some(serde_json::json!([]))),
+        ("var obj = {C:3, b_:1, c:2}", Some(serde_json::json!([]))),
+        ("var obj = {$:1, A:3, _:2, a:4}", Some(serde_json::json!([]))),
+        ("var obj = {1:1, '11':2, 2:4, A:3}", Some(serde_json::json!([]))),
+        ("var obj = {'#':1, 'Z':2, À:3, è:4}", Some(serde_json::json!([]))),
+        ("var obj = { [/(?<zero>0)/]: 1, '/(?<zero>0)/': 2 }", Some(serde_json::json!([]))), // { "ecmaVersion": 2018 },
+        // ("var obj = {a:1, b:3, [a + b]: -1, c:2}", Some(serde_json::json!([]))), // { "ecmaVersion": 6 },
```

Comment:
What happened here?

Reply:
see the PR description - there are two cases where I wasn't able to match the test cases.